src/thorlabs_mff_fastcs/controllers.py

```diff
@@ -58,6 +61,56 @@ def read_hardware_v(self, response: bytes) -> int:
 protocol = ThorlabsAPTProtocol()
 
 
+@dataclass
+class ResponseCache:
+    _last_update: datetime | None = None
+    _response: str | None = None
+    _update_event: asyncio.Event = asyncio.Event()
+
+    def __post_init__(self) -> None:
+        self._update_event.set()
+
+    def check_expired(self, time_step: float) -> bool:
+        if self._last_update is None:
+            return True
+        delta_t = datetime.now() - self._last_update
+        return delta_t.total_seconds() > time_step
+
+    def update_response(self, response: str) -> None:
+        self._response = response
+        self._last_update = datetime.now()
+
+    async def get_response(
+        self, expiry_period: float, to_await: Callable[[Any], Awaitable[str]], *args
+    ):
+        read_cache = False
+
+        # Immediately short circuit if expired
+        if self.check_expired(expiry_period):
+            # Wait if an update is in progress
+            if self._update_event.is_set():
+                self._update_event.clear()
```

Comment:
I think it would be easier to follow if you call the update response and reset the event directly here rather than relying on `read_cache` remaining `False` and doing it lower down. This should be the only path where it calls the update, I think? (Pending my other question below)

Reply:
Agree